src/yachtzee.ts

```diff
@@ -0,0 +1,831 @@
+import {
+  adv1,
+  canInteract,
+  chew,
+  cliExecute,
+  drink,
+  eat,
+  Effect,
+  fullnessLimit,
+  haveEffect,
+  inebrietyLimit,
+  itemAmount,
+  maximize,
+  myClosetMeat,
+  myFullness,
+  myInebriety,
+  myLevel,
+  myMeat,
+  mySpleenUse,
+  myTurncount,
+  numericModifier,
+  print,
+  retrievePrice,
+  spleenLimit,
+  toInt,
+  use,
+  useFamiliar,
+  useSkill,
+  visitUrl,
+} from "kolmafia";
+import {
+  $effect,
+  $familiar,
+  $item,
+  $location,
+  $skill,
+  clamp,
+  get,
+  getActiveEffects,
+  getActiveSongs,
+  have,
+  isSong,
+  Mood,
+  property,
+  set,
+  sum,
+} from "libram";
+import { acquire } from "./acquire";
+import { prepFamiliars } from "./dailies";
+import { runDiet } from "./diet";
+import { EmbezzlerFight, embezzlerSources, estimatedTurns } from "./embezzler";
+import { hasMonsterReplacers } from "./extrovermectin";
+import { baseMeat, globalOptions, safeRestore, turnsToNC } from "./lib";
+import { meatMood } from "./mood";
+import { farmingPotions, mutuallyExclusive, Potion, potionSetup } from "./potions";
+import { garboValue } from "./session";
+import synthesize from "./synthesis";
+
+class DietEntry<T> {
```

Comment:
I am extremely curious why you're setting this up here instead of in `diet.ts` and exporting from there.

Is there a compelling reason to repeat ourselves like this?

Reply:
I feel like there are some extremely large assumptions made here that are not generally applicable, the key one being that we can always solve the scheduling problem greedily - this is specific to chaining since we necessitate the excessive use of stench jellies which are of size 1.

I do acknowledge the notion that dieting functionalities should go in `diet.ts`, but I'm also of the conflicting opinion that yachtzee-specific functionalities (that aren't general and will not be used elsewhere) should go in `yachtzee.ts`.

Structurally, there's no difficulty porting the functions between files, so I'm ok with supporting either case.

Reply:
I guess my major concern here is that `DietEntry` is the name of a class in the libram diet support, and this is similar-but-not-identical to it. That feels potentially confusing. 

While I know a lot of the consumption here can be done greedily, it feels like combining this with our knapsack code might allow us to better balance potion-foods with additional yachtzees. That feels like a bit too much scope creep, then.

Reply:
Tying it in with the knapsack code is probably way out of scope imo. 
I haven't looked too deeply at the libram diet code yet, so I'm unsure whether it can currently handle balancing fishy turns with jelly charges, with consideration given to PYEC and fishy pipe as additional sources of fishy (both non-diet sources). The diet also needs to consider that using a fish juice box provides extra meat% for 30 turns since we do not need an alternative source of underwater breathing. 
That's a bunch of discrete optimisation to be conducted over multiple step functions, which all act as a soft limit (we can always get excess turns of fishy or really deep breath - they just contribute 0 profits, unlike organ fillers which are hard limits; I don't know if the knapsack code currently handles soft limits optimally).
Additionally, the diet uses marginal VOA in its optimisation, which is currently fixed as a constant by the user. However, there could be a 750MPA difference depending on how many jellies we huff (and thus how many turns of synth we can generate for the rest of embezzlies + barf). We would need to dynamically alter the user's VOA, which was originally an invariant (the user-defined VOA should have already accounted for marginal synths in a nominal garbo run, but this assumption no longer holds when we are enumerating over different amounts of forced non-synth spleen usage), to get accurate profit estimates.

Reply:
I could probably change the name `DietEntry`, but I had specifically crafted it to allow for the ease of using `.action()` for both simulation (recording the scheduled diet in the prefs) and execution (consuming and also setting jelly usage which is not tracked by mafia) by simply overriding the function. For sure, I could
1) Separate them into `.simulation()` and `.consume()` methods, and;
2a) Add a PR to libram, but once again I'm not sure if such a functionality would be generally relevant, or;
2b) Extend the `DietEntry` class to a derived `YachtzeeDietEntry` in `yachtzee.ts`, which also seems weird since I'm not really inheriting all that much from the base class.

These are all stylistic concerns, not technical ones, so if there's a general concensus / strong inclination towards a given implementation it shouldn't be too big of an issue to port things over.